src/task/camera.ts

```diff
@@ -2,8 +2,53 @@ import { type User } from '../api/auth/authorization.js';
 import { CameraModel } from '../api/db/models/Camera.js';
 import { type TaskInput } from '../api/db/models/Task.js';
 import type * as gql from '../@types/graphql.js';
+import { ProjectModel } from '../api/db/models/Project.js';
+import { DeleteImagesByFilter } from './image.js';
 
 export async function UpdateSerialNumber(task: TaskInput<gql.UpdateCameraSerialNumberInput>) {
   const context = { user: { is_superuser: true, curr_project: task.projectId } as User };
   return await CameraModel.updateSerialNumber(task.config, context);
 }
+
+export async function DeleteCamera(task: TaskInput<gql.DeleteCameraInput>) {
+  const context = { user: { is_superuser: true, curr_project: task.projectId } as User };
+
+  console.log('CameraModel.deleteCameraConfig - input: ', task.config);
+  try {
+    // Step 1: delete deployments from views
+    await ProjectModel.removeCameraFromViews(
+      {
+        cameraId: task.config.cameraId,
+      },
+      context,
+    );
+    // Step 2: delete camera record from project
+    await ProjectModel.deleteCameraConfig(
+      {
+        cameraId: task.config.cameraId,
+      },
+      context,
+    );
+
+    // Step3: delete images associated with this camera
+    await DeleteImagesByFilter({
```

Comment:
I haven't looked at the frontend code yet so I might be able to answer my own question, but it looks like we don't return the delete images task Id or do any polling here, so I'm assuming that the request from the frontend will trigger the `camera.deleteCameraTask()` mutation, the fronend will get a task Id back for this `DeleteCamera` task, perhaps do some polling for it, but meanwhile the `DeleteCamera` task is kicking off (and not awaiting the completion of) the separate `DeleteImagesByFilter` task, which may well not be completed by the time this `DeleteCamera` task is completed. Am I understanding that correctly?

If so we should think about adding polling here so we wait for the `DeleteImagesByFilter` task to complete before updating the parent `DeleteCamera` task as complete. Open to other ideas though. If we went that route, if there were errors added to the `DeleteImagesByFilter` task, I think we'd want to copy them into the `DeleteCamera` task too so that the frontend is aware of them.

Reply:
So, the way this is implemented is that that we don't actually create another task for the image deletion. I am reusing the code in the delete images task handler, but there aren't multiple tasks running at once. I do need to catch the errors with deletion and attach them to the task output here though